src/hawtio/plugins/shared/mbean/tree.ts

```diff
@@ -0,0 +1,118 @@
+import { Logger } from '@hawtio/core'
+import { escapeDots, escapeTags } from '@hawtio/util/jolokia'
+import { stringSorter } from '@hawtio/util/strings'
+import { IJmxDomain, IJmxDomains } from 'jolokia.js'
+import { MBeanNode } from './node'
+import { ILogger } from 'js-logger'
+
+export class MBeanTree {
+
+  private id: string
+  private log: ILogger
+
+  static createEmptyTree(id: string): MBeanTree {
+    return new MBeanTree(id)
+  }
+
+  static createMBeanTreeFromDomains(id: string, domains: IJmxDomains): MBeanTree {
+    const mBeanTree = new MBeanTree(id)
+    mBeanTree.populate(domains)
+    return mBeanTree
+  }
+
+  static createMBeanTreeFromNodes(id: string, nodes: MBeanNode[]) {
+    const mBeanTree = new MBeanTree(id)
+    mBeanTree.tree = nodes
+    return mBeanTree
+  }
+
+  static createFilteredTree(originalTree: MBeanNode[], filter: (node: MBeanNode) => boolean): MBeanNode[] {
+    const filteredTree: MBeanNode[] = []
+    for (const node of originalTree) {
+      const copy = node.filterClone(filter)
+      if (copy) {
+        filteredTree.push(copy)
+      }
+    }
+
+    return filteredTree
+  }
+
+  private tree: MBeanNode[] = []
+
+  private constructor(id: string) {
+    this.id = id
+    this.log = Logger.get(`${id}-tree`)
```

Comment:
We use the naming convention `hawtio-<plugin>-<area>` for loggers. Let's follow the convention.